### ISSUE TITLE
z_mergetoaddress failes with 16: bad-txns-oversize

### ISSUE THREAD

**Issue by ppratscher** [zcash/zcash#3047]
### Describe the issue
z_mergetoaddress produces oversize transactions when using shielded_limit = 0 flag.

#### If so, please list the steps to reproduce below:
1. Have a z address with several hundred unspent notes (e.g. 900)
2. Merge them to a single t1 address using the following command: zcash-cli z_mergetoaddress '["ANY_ZADDR"]' t1... 0.0001 0 0
3. z_mergetoaddress will try to merge over 100 notes in that transaction and after approximately 1h the call will fail with the 16: bad-txns-oversize error

### Expected behaviour
z_mergetoaddress should properly estimate how many notes it can include when running with the shielded_limit = 0 flag

### The version of Zcash you were using:
1.0.15

**Comment by ioptio** [comment]
We missed this ticket when you originally made it, so apologies for that. Can you try v 2.0.1 and let us know if you still have this issue?

**Comment by ppratscher** [comment]
Sorry but we have adapted our system to avoid having a large amount of unspent outputs in z addresses. Therefore we are unable to try it again with v 2.0.1

**Comment by arielgabizon** [comment]
I think your bug is related to tx size limits we used to have, at least for Sprout input addresses, I'm guessing this should work now, if you ever wanna give it a try in the future. For simplicity, you might want to wait till we integrate mergetoaddress with Sapling addresses   https://github.com/zcash/zcash/pull/3619 which should be in the next release.

**Comment by daira** [comment]
Yes, after Sapling activation the transaction size limit is 2000000 bytes rather than 100000 bytes.

On the other hand, this symptom:
> z_mergetoaddress will try to merge over 100 notes in that transaction and after approximately 1h the call will fail with the 16: bad-txns-oversize error
indicates that we're not checking what the tx size will be before making the proofs, which is clearly suboptimal even given that the larger limit is somewhat less likely to be hit.
